### PR TITLE
Multiple improvement to preprocessing and model:

### PR DESCRIPTION
**Pull request by waveworks-ai**
Multiple improvement to preprocessing and model:

 * Using frequencies up to 4200 Hz (before it was 3000 Hz)
 * Reduced feature dimensionality by averaging some spectrum frequency bins
 * Removed windows that overlap the start or end of a label, to make the training data consistent
 * Removed negative samples from very unbalanced audio files, to make the data more balanced
 * Removed class weights; no longer necessary since the data is now balanced
 * Shuffled the data properly
 * Added MaxPooling1D and Dropout to the model

In addition, the preprocessed data is saved to files, so preprocessing is only done once.

### CODE REVIEW

**Review by Breinbjerg — APPROVED** [review]
LGTM

Before you merge see my other comment in metric ESP32 project PR.

**Comment by waveworks-ai** [comment]
I have merged in your branch and made the changes I though would be enough, but it seem to be using model 1 even if I change to model 2 in the menuconfig. Can you figure out what remains?

**Comment by Breinbjerg** [comment]
Yep. Gonna take a look.

**Comment by Breinbjerg** [comment]
image

image


Added extra LOG output in `ifdefs`. Seems to compile correct code.

**Comment by waveworks-ai** [comment]
I don't think so - the input size of the second model is 672 elements, and it shouldn't take more than 4 ms to run.

**Comment by Breinbjerg** [comment]
Oh you've changed the input size. You're right.  
Gonna create a new input array for v2, and check where it's failing.

**Comment by Breinbjerg** [comment]
Sorry, had to take care of something else, but I'm back at it and it's working. 

It's been to long since I've worked with ESP32 components. I've approached it wrongly. 
Slight change in `main.py`: Changed file path to fit new component name. 
image

**Comment by waveworks-ai** [comment]
It's not building here...

`loop/esp32_metric/main/model_metric.cpp:153:10: error: 'gptimer_config_t' has no non-static data member named 'intr_priority'`

**Comment by Breinbjerg** [comment]
Interesting. Wonder why I have that on my end. 
I just added one line to remove a compile warning but it does not make any difference regarding execution. Try and remove this line. 
image

**Comment by Breinbjerg** [comment]
Changed timer api. This one is more simple and should (hopefully) not give dependencies issues.

**Comment by waveworks-ai** [comment]
I have merged to main!
